### ISSUE TITLE
unsetting an element of `list` passes all checks

### ISSUE THREAD

**Issue by someniatko** [vimeo/psalm#10042]
https://psalm.dev/r/e071179838

Expected: should trigger some issue, because a non-empty list must start with a `0` index.

Actual: as you see, no issues.

**Comment by someniatko** [comment]
for instance,

<?php

```php
$list = [ 0, 1, 2, 3, 4 ];

unset($list[0]);
var_dump($list);
```

Outputs:
```
array(4) {
  [1]=>
  int(1)
  [2]=>
  int(2)
  [3]=>
  int(3)
  [4]=>
  int(4)
}
```

which is no longer a list

**Comment by ygottschalk** [comment]
https://psalm.dev/r/06df10fb54
The error is triggered if you want to use the variable as a list.

**Comment by ygottschalk** [comment]
But not if used outside of the function -.-
https://psalm.dev/r/a70682598a
